### ISSUE TITLE
[Bug] Stmt execute failed when submit a query with hive external tbl 

### ISSUE THREAD

**Issue by dutyu** [apache/doris#20088]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.2.4

### What's Wrong?

SQL:
```sql
select 
  t_1.y_all_nums, 
  t_2.by_all_nums, 
  (t_1.y_all_nums - t_2.by_all_nums)* 100 / t_2.by_all_nums as n1 
from 
  (
    SELECT 
      'editlog_info' as name, 
      count(1) as y_all_nums 
    from 
      dp_data_db.editlog_info 
    where 
      pday = '20230416'
  ) t_1 
  left outer join (
    select 
      'editlog_info' as name, 
      count(1) as by_all_nums 
    from 
      dp_data_db.editlog_info 
    where 
      pday = '20230415'
  ) t_2 on t_1.name = t_2.name 
LIMIT 
  1000
```

Error message: `detailMessage = Fail to parse columnsFromPath, expected: [pday], filePath: hdfs://xxx/auditlog/output/20230416/part-00000`

This table is a hive external table, the tbl creater defines the source data path as `hdfs://xxx/auditlog/output/${pay}`, it does not match the "hdfs://xxx/../partition-key=partition-value" format. 

### What You Expected?

Query should not fail.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by LemonLiTree** [comment]
Is Hive Catalog?

**Comment by dutyu** [comment]
> Is Hive Catalog?

Yeah, use hive catalog, and this issue may have been fixed in master branch, is it right?

**Comment by Yulei-Yang** [comment]
it is fixed in master
